landing-pages/site/content/en/blog/testing-in-airflow-part-2-integration-tests-and-end-to-end-pipeline-tests.md

```diff
@@ -0,0 +1,16 @@
+---
+title: "Testing in Airflow Part 2 - Integration Tests and End-to-End Pipeline Tests"
+linkTitle: "Testing in Airflow Part 2 - Integration Tests and End-to-End Pipeline Tests"
+author: "Chandu Kavar and Sarang Shinde"
+twitter: "chandukavar"
+github: "chandulal"
+linkedin: "chandu-kavar-627a209b"
+description: "To know how to test the xComs and DAGs end-to-end"
+tags: ["Testing"]
+date: "2020-01-03"
+---
+In [part 1](https://blog.usejournal.com/testing-in-airflow-part-1-dag-validation-tests-dag-definition-tests-and-unit-tests-2aa94970570c) of this blog, I explained DAG validation tests, DAG definition tests, and Unit tests. These tests help to verify the syntax and cyclicity in the DAG, upstream and downstream of each task, the logic of the custom operator and sensor, etc. I would recommend reading the first part of this post.
+
+In this part, I covered the integration and end-to-end pipeline tests.
+
+You can find this blog on the medium [Testing in Airflow Part 2](https://medium.com/@chandukavar/testing-in-airflow-part-2-integration-tests-and-end-to-end-pipeline-tests-af0555cd1a82).
```

Comment:
Just to mention that as a reader, I often ran into the Medium paywall until I converted recently.
For those people who are not paying Medium customers, I think there is a way for authors to get a paywall-free link to their articles. Is it the case here ? 
Else, do you have that article on another (free) platform like for part 1 ?

Reply:
I made part2 paywall-free on medium as well.

Reply:
Let me update the free URL.

Reply:
I added a friend link for the safer side.